flyway/sql/V18__registration_funnel_1.sql

```diff
@@ -0,0 +1,49 @@
+-- for some columns we never let the caller control the contents
+CREATE OR REPLACE FUNCTION mappa.override_registration_funnel_columns()
+RETURNS TRIGGER AS $$
+BEGIN
+  IF NEW.uuid IS NOT NULL THEN
+    RAISE EXCEPTION 'You must not send uuid field';
+  ELSE
+    NEW.uuid = mappa.uuid_generate_v4();
+  END IF;
+  IF NEW.created_at IS NOT NULL THEN
+    RAISE EXCEPTION 'You must not send created_at field';
+  ELSE
+    NEW.created_at = now();
+  END IF;
+  IF NEW.email IS NULL THEN
+    RAISE EXCEPTION 'Must provide an email address';
+  END IF;
+  IF NEW.survey_results IS NULL THEN
+    RAISE EXCEPTION 'You must submit survey results';
+  END IF;
+  IF NEW.ip_address IS NULL THEN
+    RAISE EXCEPTION 'Client IP is required';
+  END IF;
+  IF NEW.registered IS NOT NULL THEN
+    RAISE EXCEPTION 'You must not send the registered field';
+  ELSE
+    NEW.registered = FALSE;
+  END IF;
+
+  RETURN NEW;
+END;
+$$ language 'plpgsql';
+
+CREATE TABLE mappa.registration_funnel_1 (
+  uuid           UUID        UNIQUE,
+  created_at     TIMESTAMPTZ,
```

Comment:
I double checked https://korban.net/postgres/book/ regarding timestamptz, and that is definitely the type we should be using. Thanks for picking the right one!